qa/workunits/rbd/rbd_mirror_stress.sh

```diff
@@ -0,0 +1,410 @@
+    rm -f $(daemon_pid_file "${cluster}")
+}
+
+admin_daemon()
+{
+    local cluster=$1 ; shift
+
+    local asok_file=$(daemon_asok_file "${cluster}" "${cluster}")
+    test -S "${asok_file}"
+
+    ceph --admin-daemon ${asok_file} $@
+}
+
+flush()
+{
+    local cluster=$1
+    local image=$2
+    local cmd="rbd mirror flush"
+
+    if [ -n "${image}" ]
+    then
+       cmd="${cmd} ${POOL}/${image}"
+    fi
+
+    admin_daemon "${cluster}" ${cmd}
+}
+
+test_image_replay_state()
+{
+    local cluster=$1
+    local image=$2
+    local test_state=$3
+    local current_state=stopped
+
+    admin_daemon "${cluster}" help |
+	fgrep "\"rbd mirror status ${POOL}/${image}\"" &&
+    admin_daemon "${cluster}" rbd mirror status ${POOL}/${image} |
+	grep -i 'state.*Replaying' &&
+    current_state=started
+
+    test "${test_state}" = "${current_state}"
+}
+
+wait_for_image_replay_state()
+{
+    local cluster=$1
+    local image=$2
+    local state=$3
+    local s
+
+    # TODO: add a way to force rbd-mirror to update replayers
+    for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
+	sleep ${s}
+	test_image_replay_state "${cluster}" "${image}" "${state}" && return 0
+    done
+    return 1
+}
+
+wait_for_image_replay_started()
+{
+    local cluster=$1
+    local image=$2
+
+    wait_for_image_replay_state "${cluster}" "${image}" started
+}
+
+get_position()
+{
+    local cluster=$1
+    local image=$2
+    local id_regexp=$3
+
+    # Parse line like below, looking for the first position
+    # [id=, commit_position=[positions=[[object_number=1, tag_tid=3, entry_tid=9], [object_number=0, tag_tid=3, entry_tid=8], [object_number=3, tag_tid=3, entry_tid=7], [object_number=2, tag_tid=3, entry_tid=6]]]]
+
+    local status_log=${TEMPDIR}/${CLUSTER2}-${POOL}-${image}.status
+    rbd --cluster ${cluster} -p ${POOL} journal status --image ${image} |
+	tee ${status_log} >&2
+    sed -nEe 's/^.*\[id='"${id_regexp}"',.*positions=\[\[([^]]*)\],.*$/\1/p' \
+	${status_log}
+}
+
+get_master_position()
+{
+    local cluster=$1
+    local image=$2
+
+    get_position "${cluster}" "${image}" ''
+}
+
+get_mirror_position()
+{
+    local cluster=$1
+    local image=$2
+
+    get_position "${cluster}" "${image}" '..*'
+}
+
+test_status_in_pool_dir()
+{
+    local cluster=$1
+    local image=$2
+    local state_pattern=$3
+    local description_pattern=$4
+
+    local status_log=${TEMPDIR}/${cluster}-${image}.mirror_status
+    rbd --cluster ${cluster} -p ${POOL} mirror image status ${image} |
+	tee ${status_log}
+    grep "state: .*${state_pattern}" ${status_log}
+    grep "description: .*${description_pattern}" ${status_log}
+}
+
+create_image()
+{
+    local cluster=$1
+    local image=$2
+    local size=$3
+
+    rbd --cluster ${cluster} -p ${POOL} create --size ${size} \
+	--image-feature exclusive-lock --image-feature journaling ${image}
+}
+
+write_image()
+{
+    local cluster=$1
+    local image=$2
+    local duration=$(($RANDOM % 35 + 15))
```

Comment:
On debian dash is used as /bin/sh, and $RANDOM is not known here.

Change `#!/bin/sh` to `#!/bin/bash`? Or use a more portable way?

https://wiki.ubuntu.com/DashAsBinSh#A.24RANDOM